seqr/utils/search/add_data_utils.py

```diff
@@ -0,0 +1,67 @@
+import elasticsearch_dsl
+
+from seqr.models import Sample
+from seqr.utils.search.constants import VCF_FILE_EXTENSIONS
+from seqr.utils.search.utils import get_es_client, get_index_metadata
+
+
+SAMPLE_FIELDS_LIST = ['samples', 'samples_num_alt_1']
+#  support .bgz instead of requiring .vcf.bgz due to issues with DSP delivery of large callsets
+DATASET_FILE_EXTENSIONS = VCF_FILE_EXTENSIONS[:-1] + ('.bgz', '.bed', '.mt')
+
+
+def validate_index_metadata_and_get_samples(elasticsearch_index, **kwargs):
```

Comment:
This function is for Elasticsearch only. It is better to put it into the Elasticsearch folder. It must be okay for Some Elasticsearch-only APIs and management commands to call the functions defined in the ES folder.

Reply:
It's not okay for anything outside the search folder to call any of the elasticsearch specific functios because it will make the switchover to a new backend and subsequent cleanup too hard. However, we can make all the function that are used outside the search folder much more generic and keep the elaticsearch specific login in the elasticsearch subfolder